x/httpproxy/path_handler.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpproxy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type pathHandler struct {
+	client http.Client
+}
+
+var _ http.Handler = (*pathHandler)(nil)
+
+func (h *pathHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.Request) {
+	requestPath := strings.TrimPrefix(proxyReq.URL.Path, "/") + "?" + proxyReq.URL.RawQuery
+	if requestPath == "" {
+		http.Error(proxyResp, "Empty path", http.StatusBadRequest)
+		return
+	}
+	targetURL, err := url.Parse(requestPath)
```

Comment:
Please rename `requestPath` to `requestURL`, since it's more than the path.
Also, you have a bug: the requestURL is never empty in your code. Make sure to test again after the change.
And let's add the query if if it's not empty.

```suggestion
	requestURL := strings.TrimPrefix(proxyReq.URL.Path, "/")
	if requestURL == "" {
		http.Error(proxyResp, "Empty path", http.StatusBadRequest)
		return
	}
	if proxyReq.URL.RawQuery != "" {
		requestURL += "?" + proxyReq.URL.RawQuery
	}
	targetURL, err := url.Parse(requestURL)
```

Reply:
With this code:
```go
requestPath := strings.TrimPrefix(proxyReq.URL.Path, "/") + "?" + proxyReq.URL.RawQuery
```

`requestPath` would always be at least `?`, so the `requestPath == ""` would never trigger.

You can test the same way you were testing before, but add a query param to the url